### PR TITLE
Add bge-m3 and Nomic 1.5 embeddings.

### PR DESCRIPTION
**Pull request by nsthorat**
Fixes https://github.com/lilacai/lilac/issues/1148

I added two versions of the Nomic 1.5 embeddings, which use matryoshka_dim dims, see: https://huggingface.co/nomic-ai/nomic-embed-text-v1.5

Demo: https://lilacai-nikhil-staging.hf.space/datasets#local/glue_ax_map&schemaCollapsed=false

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -86,6 +89,7 @@ hdbscan = { version = "^0.8.33", optional = true }
 
 [tool.poetry.extras]
 all = [
+  "bge",
```
Comment (dsmilkov) [inline pyproject.toml:92]: these have to be pypi packages, not aliases to groups

File: lilac/embeddings/nomic_embed.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      raise ImportError(
+        'Could not import the "sentence_transformers" python package. '
+        'Please install it with `pip install "sentence_transformers".'
+      )
+
+    def _encode(doc: list[str]) -> list[np.ndarray]:
+      embeddings = self._model.encode(doc, convert_to_tensor=True)
+      # Extract the dense vectors from the model.
+      embeddings = F.layer_norm(embeddings, normalized_shape=(embeddings.shape[1],))
+      embeddings = embeddings[:, : self._matryoshka_dim]
```
Comment (dsmilkov) [inline lilac/embeddings/nomic_embed.py:73]: are dimensions sorted by importance , like PCA ? curious why we can just take the last 256/512 dimensions

**Review by dsmilkov — APPROVED** [review]
nice!

File: lilac/embeddings/nomic_embed.py (hunk 2 above)
Comment (nsthorat) [reply lilac/embeddings/nomic_embed.py:73]: thats the magic of matryoshka_dim.. it's not PCA exactly, afaict it's baked into the loss function: https://aniketrege.github.io/blog/2024/mrl/

File: pyproject.toml (hunk 1 above)
Comment (nsthorat) [reply pyproject.toml]: done

**Pull request merged**